src/Data/Food/Product.elm

```diff
@@ -0,0 +1,618 @@
+
+
+isTransport : ProcessName -> Bool
+isTransport (ProcessName processName) =
+    String.startsWith "Transport, " processName
+
+
+isIngredient : ProcessName -> Bool
+isIngredient processName =
+    (isProcessing processName
+        || isWaste processName
+        || isTransport processName
+    )
+        |> not
+
+
+type alias Process =
+    { amount : Amount
+    , impacts : Impacts
+    }
+
+
+type alias ImpactsForProcesses =
+    AnyDict String ProcessName Impacts
+
+
+emptyImpactsForProcesses : ImpactsForProcesses
+emptyImpactsForProcesses =
+    AnyDict.empty processNameToString
+
+
+type alias Processes =
+    AnyDict String ProcessName Process
+
+
+emptyProcesses : Processes
+emptyProcesses =
+    AnyDict.empty processNameToString
+
+
+insertProcess : ProcessName -> Process -> Processes -> Processes
+insertProcess name process processes =
+    AnyDict.insert name process processes
+
+
+computeProcessPefImpact : List Definition -> Processes -> Processes
+computeProcessPefImpact definitions processes =
+    processes
+        |> AnyDict.map
+            (\_ process ->
+                { process
+                    | impacts =
+                        Impact.updatePefImpact definitions process.impacts
+                }
+            )
+
+
+findImpactsByName : ProcessName -> ImpactsForProcesses -> Result String Impacts
+findImpactsByName ((ProcessName name) as procName) =
+    AnyDict.get procName
+        >> Result.fromMaybe ("Procédé introuvable par nom : " ++ name)
+
+
+decodeProcesses : List Definition -> Decoder ImpactsForProcesses
+decodeProcesses definitions =
+    AnyDict.decode (\str _ -> ProcessName str) processNameToString (Impact.decodeImpacts definitions)
+
+
+
+---- Step
+
+
+type alias Step =
+    { ingredients : Processes
+    , transport : Processes
+    , waste : Processes
+    , processing : Processes
+    }
+
+
+type alias Product =
+    { consumer : Step
+    , supermarket : Step
+    , distribution : Step
+    , packaging : Step
+    , plant : Step
+    }
+
+
+type ProductName
+    = ProductName String
+
+
+productNameToString : ProductName -> String
+productNameToString (ProductName name) =
+    name
+
+
+stringToProductName : String -> ProductName
+stringToProductName str =
+    ProductName str
+
+
+type alias Products =
+    AnyDict String ProductName Product
+
+
+emptyProducts : Products
+emptyProducts =
+    AnyDict.empty productNameToString
+
+
+emptyStep : Step
+emptyStep =
+    { ingredients = emptyProcesses
+    , transport = emptyProcesses
+    , waste = emptyProcesses
+    , processing = emptyProcesses
+    }
+
+
+type alias Ingredient =
+    ( ProcessName, Unit.Ratio )
+
+
+type alias ProductDefinition =
+    { consumer : List Ingredient
+    , supermarket : List Ingredient
+    , distribution : List Ingredient
+    , packaging : List Ingredient
+    , plant : List Ingredient
+    }
+
+
+type alias WeightRatio =
+    { processName : ProcessName
+    , weightRatio : Float
+    }
```

Comment:
What the type alias `WeightRatio` is storing is the name of the process that's the heaviest ingredient (see the comment about this hack at https://github.com/MTES-MCT/ecobalyse/pull/128/files#diff-a93cff7f52063f142226bed3613428b025be21f94e27991a54140bfbc8e28655R471-R474), and the "raw/cooked" ratio that we need to compute the impacts .

The name is a bit confusing, so let me try something along the lines of: 
```
type alias RawCookedRatioInfo =
    { weightLossProcessName : ProcessName
    , rawCookedRatio = Float
    }
```

And just to clarify, this type is storing different information than the type `Ingredient` which is storing a process name an its amount needed for a recipe.

Reply:
> (see the comment about this hack at https://github.com/MTES-MCT/ecobalyse/pull/128/files#diff-a93cff7f52063f142226bed3613428b025be21f94e27991a54140bfbc8e28655R471-R474)

Unfortunately this link doesn't point to a comment anymore as the associated code has changed

Anyway, side question: what would `rawCookedRatio` to be a `Unit.Ratio`?

Reply:
Sure, why not. Not quite sure yet what we would gain by that, but i trust you on that one, and maybe at some point you can explain on zoom ;)

Reply:
The general benefit is to prevent performing operations on mismatched data types, and get appropriate compilation errors when trying to do so. Would happily explain it over a zoom!